### ISSUE TITLE
html version of 'new' OMNI template

### ISSUE THREAD

**Issue by OskaratOmni** [rfortherestofus/omni#89]
Hey @tvroylandt  and @dgkeyes  
We have updated versions of the OMNI template for Word and for pdf, but not for html yet. 

Do you think you could go ahead make the html one? The main feature to try to implement is the header. 
Much of the material in the style guide (the one I sent months ago when you made the pdf one) won't be relevant or necessary. But please go ahead and give that a shot. 

Thanks! 
Oskar

**Comment by dgkeyes** [comment]
Sure, we can work on this. I'm probably going to bring in Joseph, a new consultant, to help with this. I need to confirm his availability. I'll get back to you soon on this.

**Comment by dgkeyes** [comment]
Ok, Joseph is able to work on this. @JosephBARBIERDARNAL please meet Oskar (@OskaratOmni). Oskar, Joseph has been doing a lot of similar work to Thomas for us (data viz, reports, etc). 

@JosephBARBIERDARNAL I recorded a video to explain this: https://show.rfor.us/97tSQ1m6. If you have any questions, please let me know.

**Comment by JosephBARBIERDARNAL** [comment]
Hello Oskar!

I've opened a draft here (#90) for this issue

**Comment by OskaratOmni** [comment]
@JosephBARBIERDARNAL , thanks! Looks like great progress has been made. @dgkeyes  let me know when you think I should re-load the OMNI package on our end for a little test driving.

**Comment by dgkeyes** [comment]
@OskaratOmni feel free to test at any point. The remaining tweaks @JosephBARBIERDARNAL talked about in the PR (#90) are minor so you should be able to use the template fine. When you/others test, make sure to use this template:

![Uploading CleanShot 2024-11-12 at 16.45.48.png…]()

**Comment by OskaratOmni** [comment]
Hey @dgkeyes , for some reason I'm not seeing that template in my drop-down. 
I have run 
remotes::install_github("rfortherestofus/omni", force = TRUE)
a couple of times and restarted R but am not seeing OMNI RMD Template. 
Anything I should do differently on my end?

**Comment by OskaratOmni** [comment]
Thanks @dgkeyes. That worked and I can indeed see the template now. 
When I knit, it looks like this: 
![image](https://github.com/user-attachments/assets/90a6a785-55b3-4ced-a84b-3dc6d8feef77)

In case it got lost in the months that have past since we've revamped our template, we are hoping for something broadly similar to what is in the OMNI pdf new: 
![image](https://github.com/user-attachments/assets/b53bd6c9-5a1d-4893-8055-1f3149268b2a)

**Comment by dgkeyes** [comment]
That's weird. When I tested it in the package, it looks like this:

![image](https://github.com/user-attachments/assets/65a68704-2720-4129-800f-43f3a0d28d96)

But now that I'm testing it in a different project, I see what you're seeing.

**Comment by OskaratOmni** [comment]
Oh interesting. Something about location of css? I definitely have some packages that are due for an update but I'm not sure that's it.

**Comment by dgkeyes** [comment]
No, I think I see the issue in the code: the CSS and image files are referred to in a way that only works in the package itself. @JosephBARBIERDARNAL was actually mentioning this in the PR but I didn't quite get the issue. I pushed a quick change that makes this work. You can test it out using `remotes::install_github("rfortherestofus/omni", force = TRUE, ref = "fix-assets-in-html")`.

**Comment by dgkeyes** [comment]
@tvroylandt what is the best approach here for the long term to use assets from the package in the template.html file that @JosephBARBIERDARNAL created? If it's within R code, I would use `system.file()` but I'm not sure the best approach in an HTML file.

**Comment by OskaratOmni** [comment]
That makes sense. I was thinking of that note by @JosephBARBIERDARNAL too

**Comment by OskaratOmni** [comment]
That's working now, thanks!

**Comment by JosephBARBIERDARNAL** [comment]
Glad to see you've found a solution!

**Comment by dgkeyes** [comment]
Hey @OskaratOmni just wanted to check in on this. How's it working? Any changes needed?

**Comment by OskaratOmni** [comment]
Hey @dgkeyes  - sorry, haven't played with it too much  yet. But I'll do that later this week and I asked a few others on the team to look at it too. At a glance it looks great.

**Comment by dgkeyes** [comment]
> Would it be possible to adjust the assets call so it works like the other templates?

Oh yes, forgot about that. I'll get back to you on it.

**Comment by tvroylandt** [comment]
Ok. I would say the best way to handle that is :

- wrap the html and css call into an R format
- leave the logo ref to a URL

**Comment by JosephBARBIERDARNAL** [comment]
> * wrap the html and css call into an R format

Not sure what you mean by that. Like calling r\`system.file()` for calling the css?

**Comment by tvroylandt** [comment]
Something like this : https://github.com/rfortherestofus/omni/blob/main/R/omni_formats.R#L166

**Comment by dgkeyes** [comment]
@JosephBARBIERDARNAL do you want to give this a shot?

**Comment by JosephBARBIERDARNAL** [comment]
@dgkeyes yes I'll do it!

**Comment by dgkeyes** [comment]
@OskaratOmni I set this to closed but if you need us to do anything else on it, please feel free to reopen.

**Comment by OskaratOmni** [comment]
Hey @dgkeyes I just wanted to flag that this template is not working. 
I think the issue is partly how the template loads into a sub-folder, but if I fix that it still doesn't have the formatting that it does before. 

However, other more complicated news is that OMNI is going to do a fairly major rebrand - this will almost certainly change the color scheme and might change the logo and even fonts and stuff. 

Given that, could we please pull this template from the package for now? I was trying to use it to share with a client later in the week and I just couldn't get it to work. But if we were to fix it, it might be outdated in just a few weeks due to the impending changes in style guidance. I think that the other templates we have will hold us over in the interim. 

Thanks!

**Comment by dgkeyes** [comment]
Sure, no problem @OskaratOmni. I'm going to create a new issue for this so I can make a branch and have you ensure that everything works. 

Let me know about the rebrand. I love a good rebrand!
